src/requirements.cpp

```diff
@@ -173,6 +173,23 @@ void requirement_data::load_obj_list( JsonArray &jsarr, std::vector< std::vector
     }
 }
 
+requirement_data requirement_data::operator*( unsigned scalar ) const
+{
+    requirement_data res = *this;
+    for( auto &group : res.components ) {
+        for( auto &e : group ) {
+            e.count = std::max( e.count *= scalar, -1 );
```

Comment:
Why are you writing to `e.count` inside the `std::max` *and* assigning to it afterwards? Same below.